docs/source/tutorials/SAR/SAR tutorial.ipynb

```diff
@@ -0,0 +1,1180 @@
+   "id": "a350c00b",
+   "metadata": {},
+   "source": [
+    "`get_activations()` extracts muscle activation data at each timestep of the trained policy's rollouts. In this implementation, `get_activations()` saves muscle activations from a given rollout if one of two conditions is met: (a) the task is solved during that rollout, or (b) the rewards from that rollout fall above a certain threshold, computed as a percentile from a set of preview rollouts. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3be60e30",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_activations(name, env_name, seed, episodes=1000, percentile=80):\n",
+    "    \"\"\"\n",
+    "    Returns muscle activation data from N runs of a trained policy.\n",
+    "\n",
+    "    name: str; policy name (see train())\n",
+    "    env_name: str; name of the gym environment\n",
+    "    seed: str; seed of the trained policy\n",
+    "    episodes: int; optional; how many rollouts?\n",
+    "    mode: str; optional; 'solved' or 'pos_reward'\n",
+    "    percentile: int; optional; percentile to set the reward threshold for considering an episode as successful\n",
+    "    \"\"\"\n",
+    "    with gym.make(env_name) as env:\n",
+    "        env.reset()\n",
+    "\n",
+    "        model = SAC.load(f'{name}_model_{env_name}_{seed}')\n",
+    "        vec = VecNormalize.load(f'{name}_env_{env_name}_{seed}', DummyVecEnv([lambda: env]))\n",
+    "\n",
+    "        # Calculate the reward threshold from 100 preview episodes\n",
+    "        preview_rewards = []\n",
+    "        for _ in range(100):\n",
+    "            env.reset()\n",
+    "            rewards = 0\n",
+    "            done = False\n",
+    "            while not done:\n",
+    "                o = env.get_obs()\n",
+    "                o = vec.normalize_obs(o)\n",
+    "                a, __ = model.predict(o, deterministic=False)\n",
+    "                next_o, r, done, info = env.step(a)\n",
+    "                rewards += r\n",
+    "            preview_rewards.append(rewards)\n",
+    "        reward_threshold = np.percentile(preview_rewards, percentile)\n",
+    "\n",
+    "        # Run the main episode loop\n",
+    "        solved_eps = []\n",
+    "        solved_acts = []\n",
+    "        for i, _ in tqdm(enumerate(range(episodes))):\n",
+    "            env.reset()\n",
+    "            rewards, is_solved, acts = 0, [], []\n",
+    "            done = False\n",
+    "            while not done:\n",
+    "                o = env.get_obs()\n",
+    "                o = vec.normalize_obs(o)\n",
+    "                a, __ = model.predict(o, deterministic=False)\n",
+    "                next_o, r, done, info = env.step(a)\n",
+    "                is_solved.append(info['solved'])\n",
+    "                acts.append(env.sim.data.act.copy())\n",
+    "\n",
+    "                rewards += r\n",
+    "            if 'Leg' in env_name:\n",
```

Comment:
ok, this looks like a hack now. It will be better to give `bool check_solved=True` as the input parameter to the function.